### PR TITLE
`pick-to-branch`: warn if target remote is not in `github.openssl.org`

### LINKED ISSUE #141: Commits pushed by `pick_to_branch` tend to get lost
Since a couple of months it happens to me rather often 
(for instance, today for https://github.com/openssl/openssl/pull/20298 and https://github.com/openssl/openssl/pull/20257) that 
after a successful `pick-to-branch` the commits do not really end up (or at least do not prevail) in the upstream destination.

As a workaround, I have to check after a couple of minutes and in case the commits were silently ignored, push them manually.

### PR DESCRIPTION
**Pull request by DDvO**
Fixes #141 by adding a warning when needed

### CODE REVIEW

File: review-tools/pick-to-branch [hunk 1]
```diff
@@ -69,8 +69,18 @@ m*)
...
 esac
 
+REMOTE=`git branch --list $TARGET -vv | sed 's/^\* //' \
+ | awk '{ sub(/\[/, "", $3); sub(/\/.*/, "", $3);  print $3 }'`
+# usually this will be 'upstream'
+if [ "$REMOTE" = "" ] ; then
+    echo Cannot find git remote for target branch $TARGET
+    exit 1
+fi
+git remote get-url $REMOTE | grep -q -E 'github.openssl.org:(openssl|omc|otc)' \
```
Comment (t8m) [inline review-tools/pick-to-branch:79]: Please drop the `(openssl|omc|otc)` part from the regexp. No need to be too restrictive.

Comment (DDvO) [reply review-tools/pick-to-branch]: Done

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

File: review-tools/pick-to-branch [hunk 2]
```diff
@@ -69,8 +69,18 @@ m*)
     ;;
 esac
 
+REMOTE=`git branch --list $TARGET -vv | sed 's/^\* //' \
+ | awk '{ sub(/\[/, "", $3); sub(/\/.*/, "", $3);  print $3 }'`
```
Comment (levitte) [inline review-tools/pick-to-branch:73]: The more git-ish way to do this seems to be:

``` shell
REMOTE=$(git for-each-ref --format='%(push:remotename)' \
                  $(git symbolic-ref -q HEAD))
```

Comment (DDvO) [reply review-tools/pick-to-branch:73]: Nice, but this does not detect the remote of `$TARGET`,
and replacing `HEAD` by  `$TARGET` does not work.

Comment (levitte) [reply review-tools/pick-to-branch:73]: You're right...  but `git rev-parse` to the rescue!

``` shell
REMOTE=$(git for-each-ref --format='%(push:remotename)' \
             $(git rev-parse --symbolic-full-name $TARGET))
```

Comment (DDvO) [reply review-tools/pick-to-branch]: Thanks - this works; improvement done.

**Comment by DDvO** [comment]
@t8m and @paulidale thanks for your approvals;
please re-approve after the improvement suggested by @levitte.

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by DDvO** [comment]
Thanks @levitte for the improvement and @t8m and @paulidale for the approvals.
Looks like this is ready to merge.

**Pull request closed without merging**